treescript/src/treescript/github/client.py

```diff
@@ -83,10 +82,26 @@ async def commit(self, branch: str, message: str, additions: Optional[Dict[str,
                 },
                 "message": {"headline": message},
                 "fileChanges": changes,
-                "expectedHeadOid": head_oid,
+                "expectedHeadOid": None,
             }
         }
-        await self._client.execute(query, variables=variables)
+
+        # Retry the query a few times in-case the head_oid was changed.
+        async def _execute():
+            head_oid = (await self._client.execute(oid_query))["repository"]["object"]["oid"]
+            variables["input"]["expectedHeadOid"] = head_oid
+            await self._client.execute(commit_query, variables=variables)
+
+        retries = 2
+        while True:
+            try:
+                await _execute()
+                break
+            except TransportQueryError:
+                if retries <= 0:
+                    raise
+                log.debug("Head rev didn't match expected, retrying")
+                retries -= 1
```

Comment:
Refactor: Good news! `scriptwoker_client` already provides a decorator that retries async functions! E.g.: 

https://github.com/mozilla-releng/scriptworker-scripts/blob/89a90648f3710c9ffab062bc9207af84a3e0cfbe/githubscript/src/githubscript/github.py#L47-L53

Reply:
Nice, updated!